File: 0601_0700/0688/0688.py
def dist(a, b):
    return ((a[0] - b[0]) * (a[0] - b[0]) + (a[1] - b[1]) * (a[1] - b[1])) ** 0.5

with open("input.txt", "r") as f, open("output.txt", "w") as q:
    g = tuple(int(x) for x in f.readline().split())
    d = tuple(int(x) for x in f.readline().split())
    n = int(f.readline())
    i = 1
    while i <= n:
        t = tuple(int(x) for x in f.readline().split())
        if (2 * dist(g, t) <= dist(d, t)):
            break
        i += 1
    q.write("NO" if i > n else f"{i}")

File: 0601_0700/0688/0688.c
#include <stdio.h>
#include <math.h>

typedef struct {int x, y;} dot;

double distance(dot a, dot b)
{
	return sqrt((double)((a.x - b.x) * (a.x - b.x) + (a.y - b.y) * (a.y - b.y)));
}

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int i, n;
	dot d, g, t;

	fscanf(f,"%d%d%d%d%d", &g.x, &g.y, &d.x, &d.y, &n);
	for (i = 1; i <= n; ++i)
	{
		fscanf(f, "%d%d", &t.x, &t.y);
		if (2 * distance(g, t) <= distance(d, t))
			break;
	}
	fprintf(q, i > n ? "NO" : "%d", i);
	return 0;
}
